Fix imap4-utf-7 codec lookup function for Python 3.9

mgorny opened the pull request:
Python 3.9 normalizes the codec name into 'imap4_utf_7' rather than
'imap4-utf-7', and therefore the lookup function needs to account
for the former name.  Transform the latter locally to preserve support
for all Python versions.

Fixes: ticket: 9832

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9832
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
  * am I supposed to fix coding style issues around my change?
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [ ] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

wiml reviewed (approved):
LGTM. Added some notes to the Trac bug.

rodrigc reviewed (approved):
This is OK

rodrigc committed:
Fix imap4-utf-7 codec lookup function for Python 3.9

Python 3.9 normalizes the codec name into 'imap4_utf_7' rather than
'imap4-utf-7', and therefore the lookup function needs to account
for the former name.  Transform the latter locally to preserve support
for all Python versions.

Submitted by:  Michal Gorny <mgorny@gentoo.org>

rodrigc commented:
@mgorny This is a very interesting thing.  It looks like because your name has non-ASCII characters in it: Michał Górny , when GitHub sends your PR to Azure Pipelines to run the Twisted tests, somehow some detail about your name makes it into the Windows environment variables.  This is why your PR was failing on Windows, even though you did a very minor change which should not have affected anything.

@hawkowl figured this out and posted the details here: https://twistedmatrix.com/pipermail/twisted-python/2020-June/065140.html

For now I worked around the problem by amending your commit to have my name as the author, but I put your name in the commit log message (without non-ASCII characters).

Sorry for the inconvenience!
Thanks for submitting fixes to Twisted!

mgorny commented:
Heh, and I thought that the times of non-ASCII names breaking everything are over ;-).
